kernel/proc/scheduler.md

```diff
@@ -0,0 +1,30 @@
+# Kernel - Processes and threads - Scheduler
+The operating system scheduler controls threads execution with a predetermined policy. It is a part of Phoenix-RTOS having the most significant influence on the performance and responsiveness of the whole system.
+
+## Scheduling policy
+The scheduling algorithm is defined in the `threads_schedule` function, which is invoked by a timer hardware interrupt. Each function's invocation changes the context of the currently executed thread to the next one, available in a thread's ready queue. The scheduling policy determined which thread shall be chosen for the next execution. In Phoenix-RTOS, there is used a priority preempted round-robin algorithm. There are eight priority levels, the smallest value holds the highest priority.
+
+
+The thread management unit contains an eight-element array, where each of them holds a pointer to the list of threads of the same priority. A scheduling algorithm is defined as follows:
+
+1. The spinlock is set before any operations on common thread management data.
+2. The unit holds information about the running threads of each processor's core. The current thread's context for the interrupted core is saved and added to the end of the priority list.
+3. The next available thread with the highest priority is selected to be run next and is removed from the ready thread's list. If a selected thread is a ghost, it is added to the ghosts list and the reaper thread is waked up.
```

Comment:
Repetition. We need to clarify what a ghost is (it's not a common concept). Perhaps some reference to a place in documentation that explains this?
```suggestion
3. The next available thread with the highest priority is selected to be run and is removed from the ready thread's list. If a selected thread is a ghost, it is added to the ghosts list and the reaper thread is waked up.
```